pychromecast/__init__.py

```diff
@@ -171,8 +179,23 @@ def __init__(self, host, port=None, device=None, **kwargs):
         self.status = None
         self.status_event = threading.Event()
 
+        self.browser = None
+        self.tracker = None
+
+        self.setup(host, port, self.device, self.kwargs)
+
+    def setup(self, host, port, device, kwargs):
+        """
+        Method for setting up, and resetting the chromecast object
+        """
+        tries = 2 if self.cast_type == CAST_TYPE_GROUP \
+            else kwargs.pop('tries', None)
+        timeout = kwargs.pop('timeout', None)
+        retry_wait = kwargs.pop('retry_wait', None)
+        blocking = kwargs.pop('blocking', True)
+
         self.socket_client = socket_client.SocketClient(
```

Comment:
As the number of tries is a finite number the old socket client will not try to reconnect forever, also it is stopped [here](https://github.com/balloob/pychromecast/pull/167/commits/d57f10626870df96401e98f579ad35b6653d9405#diff-cd08193135e2ec94e86f3678d2bfdddfR422)